src/Common/Acl/AclExtended.php

```diff
@@ -236,8 +237,11 @@ public static function setUserAro($arr_group_titles, $user_name, $first_name, $m
         //see if this user is gacl protected (ie. do not allow
         //removal from the Administrators group)
         require_once(dirname(__FILE__) . '/../../../library/user.inc');
-        require_once(dirname(__FILE__) . '/../../../library/calendar.inc');
-        $userNametoID = getIDfromUser($user_name);
+
+        $userService = new UserService();
+        $user = $userService->getIdByUsername($pc_username);
```

Comment:
seems odd that parameter was $user_name in prior call, but $pc_username in new call?